backend/hitas/models/condition_of_sale.py

```diff
@@ -55,6 +56,41 @@ class Meta:
     def fulfilled(self) -> Optional[datetime.datetime]:
         return self.deleted  # noqa
 
+    def get_first_purchase_date(self) -> Optional[datetime.date]:
+        """
+        Return the first purchase date of the new apartment.
+        Can be used to determine the sell by date.
+        When apartment was sold after it was completed, the sell by date is calculated based on first sale date.
+        """
+        from hitas.models import ApartmentSale
+
+        return (
+            ApartmentSale.objects.filter(apartment=self.new_ownership.apartment)
+            .order_by("purchase_date")
+            .values_list("purchase_date", flat=True)
+            .first()
+        )
+
+    @property
+    def sell_by_date(self) -> Optional[datetime.date]:
+        # If the apartment has not been completed, there is no sell by date yet.
+        sell_by_date = self.new_ownership.apartment.completion_date
+        if sell_by_date is None:
+            return None
+
+        # If apartment was sold after it was completed, the sell by date is calculated based on first sale date.
+        first_purchase_date = getattr(self, "first_purchase_date", self.get_first_purchase_date())
```

Comment:
You should prefer the `hasattr` guard clause over `getattr` (even though it is more verbose), as `getattr` evaluates the default argument as well, making another query when you're trying to save them here.

Reply:
Good catch! I had already fixed in the other sell_by_date method 😄